netutils/netinit/netinit.c

```diff
@@ -533,6 +554,10 @@ static void netinit_configure(void)
 #ifndef CONFIG_NETINIT_NETLOCAL
   /* Bring the network up. */
 
+#ifdef CONFIG_NETINIT_DHCPC
+    memset(&g_ds, 0, sizeof(g_ds));
```

Comment:
don't need, global variable is already zerod

Reply:
Sorry for the long, somewhat tangent reply.

I was curious about this comment. While I agree it is not necessary for the normal call vector, who says that `netinit_configure()` cannot be called a second time. In this case, I'd think ensuring a known state is a good idea.

I was curious if it was called anywhere twice, so I searched where `netinit_configure()` is called from. The only place it is _currently_ called is `netinit_bringup()` or `netinit_thread()` (which `netinit_bringup()` launches) 

But where is `netinit_bringup()` called from? 
`nsh_consolemain()`
Looking at the documentation I see:

``` 
 * Name: nsh_consolemain (Normal character device version)
 *
 * Description:
 *   This interfaces may be to called or started with task_start to start a
 *   single an NSH instance that operates on stdin and stdout.  This
 *   function does not normally return (see below).```

I think this raises a concern. If I wanted multiple NSH consoles, does this mean that my network gets re-configured each time any of them come up?

Reply:
There are three nsh_consolemain() functions, each of them calls netinit_bringup() unconditionally.  There is only one system console, but potentially many telnet NSH sessions.  There are at least two ways to get multiple session:

- Telnet.  But in this case, nsh_consolemain() will not be called.  A special nsh_telnetmain() is called instead and it does not re-initialize the network.  
- You can also create a new NSH session from the `nsh>` command line.  This, I suspect will be a problem because I think that nsh_consolemain() (and, hence, netinit_bringup()) WILL be called again.

I don't see any protections in netinit_bringup() or netinit_configure() to make them idempotent.  What does happen in this case?  My guess is that you are correct to be concerned.

Reply:
@antmerlino  Yes that make sense, I was only looking at the init thread case and overlooked it.